### ISSUE TITLE
Can't go directly to fullscreen when tapping thumbnail

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#4152]
### Version

- 0.20.0

### Regression
The user can no longer go into fullscreen directly upon tapping the thumbnail in the video details page, like in 0.19.8. (for those checking: requires locked system rotation and landscape mode in main player)

### Additional Note

MUH FULLSCREEN!

**Comment by opusforlife2** [comment]
I did give STR: "(for those checking: requires locked system rotation and landscape mode in main player)"

> Btw, isn't it a duplicate of #4070?

That's the issue I was browsing when I remembered I wanted to open this one. That one is for external links. This one is for the thumbnail in video details.

**Comment by EliasKomar** [comment]
I think this is a good idea. It feels more natural to open the video upon clicking the thumbnail, than opening a info page from where you can start the video. It feels more natural (maybe just because it's what youtube does).

**Comment by opusforlife2** [comment]
@EliasKomar This issue is for the thumbnail you see in the video details page, not for thumbnails in video lists like search results, playlists, etc. If you want the video to autoplay upon opening like Youtube does, then that's the default behaviour in the upcoming unified player.

@B0pol Sorry, this comment made me realise I didn't specify _where_ the thumbnail was, which led to your questions. I'll edit the OP.

**Comment by Stypox** [comment]
If the fullscreen opened when tapping the thumbnail, it would be an inconsistent and possibly unexpected behaviour with respect to how the video starts playing with autoplay enabled. But at the same time I think the benefits outcome the cons, so I agree this should be changed. @avently what do you think?

**Comment by opusforlife2** [comment]
Well, yeah. You can only see the thumbnail when autoplay is off. Otherwise, when autoplay is on, the video is already playing and it's one tap to go full screen. If autoplay is off, it's two taps: one to play the video in portrait mode, and another to go full screen.

**Comment by avently** [comment]
@Stypox 
>what do you think?

I think it's a bad idea. We already discussed it in the unified PR:
- making the video automatically rotated to landscape from portrait on play looks ugly
- inconsistency everywhere.

You probably didn't see how it looks like but I did. I use this for more than 1 year long. I added this "mode" just because theScrabi asked about it in my first PR here.

I'm ok if it will be optional and disabled by default but even in this case I will never write any line of code for this "anti-feature".

Yeah, it may sound unpleasant but it's how I feel like when I remember this ugly transition from portrait to landscape on every play.

I just use autorotation enabled globally and the only thing I need to do is to rotate the phone for viewing on fullscreen.

**Comment by opusforlife2** [comment]
@avently Are you talking about the transition that exists in 0.19.8? Or was yours different in someway?

**Comment by avently** [comment]
> Are you talking about the transition that exists in 0.19.8? Or was yours different in someway?

I'm talking about transition that takes place after clicking on fuillscreen button. If the feature you want will be realized than the same action will be made after a stream starts playing. And this is bad in terms of usability. If you want the rotation happens just press on the fullscreen button. It's ok because you really expect that the screen rotates.

**Comment by opusforlife2** [comment]
>  If the feature you want will be realized than the same action will be made after a stream starts playing.

Yes, that's what happens in 0.19.8 and all the releases before it, and it's the behaviour I'm used to.

Since the autoplay option didn't exist before, it wasn't a problem. But now that it does, this should be a setting, dependent on the autoplay one. So if autoplay is on, this setting should be greyed out.

But I'm concerned about this:
> inconsistency everywhere

Are you talking about bugs appearing elsewhere because of this code change?

**Comment by avently** [comment]
> Are you talking about bugs appearing elsewhere because of this code change?

No, it will not produce any bugs, it's just a couple of lines of code but it doesn't look logical choice to rotate the screen after pressing on thumbnail.

**Comment by opusforlife2** [comment]
Ah well, it would be behind a toggle, so those who don't want it can avoid it.

**Comment by avently** [comment]
Guys, when do you plan to release 0.20.0? Looks like some PR should be merged and we are done?

**Comment by Stypox** [comment]
@avently this Sunday @TobiGr is planning to publish a RC, and these PRs need to be merged #4246, #4254, (#4255, but probably not since there is also your solution, we'll see), #4259, #4272, #4288, #4310

**Comment by nikhilCad** [comment]
Please make it an opt in and disable it by default.
Something like:
Default- current behaviour
Landscape- turns autoplay off, opens in landscape

**Comment by avently** [comment]
You are talking about fullscreen after play, it can be easily implemented but the problem I see is when the device is in portrait mode. So what do you expect to happen after play when your device is in portrait and horizontal video started to play?
• device stays in the same orientation but opens on fullscreen
• device switches to landscape and opens in fullscreen?

Automatic action can only be made when you have global autorotation disabled (I.e. locked to portrait or landscape).

And a part of the question, what should happen when you stopped watching the video (I.e. video finished or you pressed back button)?

What do you expect to see when the global autorotation is enabled?
• device should go into fullscreen
• nothing should happen?

The problem is these questions I don't have universal answers for. Because some people would want one behavior, another people want other behavior. From my point of view, the app shouldn't go into fullscreen when the rotation is not appropriate for the video. For example, when the device in portrait (with locked orientation) playing horizontal video in fullscreen should only be possible after automatic rotation to landscape. Same for vertical videos: fullscreen should only be possible when device already rotated or rotated automatically to portrait. The autorotation i'm talking about is already implemented (on phones) and you can test how it works in playlists with mixed horizontal and vertical videos: in locked orientation in fullscreen your phone will be rotated to different orientations appropriate for a video orientation.

**Comment by nikhilCad** [comment]
I personaply prefer this(do note that in comment above I said it should be
opt-in and disabled by default):
I click on the video, its video info shows with autoplay turned off, I
again click on it, it stars playing fullscreen+landscape(irrespective of
phone rotation) , When I click back it should go back to portrait(phone'
rotation) and show video info.
(To be fair I hate video info too, it wastes time and one unnecessary click
gets added)
Anyways it is my personal preference and that is why I think it should be
made configurable in settings(and disabled by default).
I will try(try being a key word) to submit a PR for the same this week.

**Comment by opusforlife2** [comment]
@avently The automatic fullscreening behaviour doesn't need to change from the current one at all. Pretend that the feature is implemented such that after the user taps the thumbnail to play, the app immediately auto-taps the full screen button. That's it. Whatever behaviour the app has currently, can continue to be so.

The full screen button should behave as it does now.
The back button should behave as it does now.

> And a part of the question, what should happen when you stopped watching the video (I.e. video finished or you pressed back button)?

Personally, I think the player should remain in fullscreen and wait for the user to tap Back to go back to video details. But this can be tested properly only after the basic auto-full screen behaviour is in place. So it doesn't need to be part of the initial commit/effort. We'll get more knowledge about what looks good or bad when we have an apk to test.

> What do you expect to see when the global autorotation is enabled?

Nothing should change there, I think. Because full screen is a function of rotating the phone in that case, it shouldn't change anything in the current behaviour. No extra tap required there like it is in locked orientation.

> You are talking about fullscreen after play, it can be easily implemented but the problem I see is when the device is in portrait mode.

🤭 We only need auto full screen for portrait mode. The app automatically goes into full screen if you tap the thumbnail in locked landscape rotation.

**Comment by johanw666** [comment]
I agree fully with @sanityormadness . TThe use case "watching video in bed" is a large one for me, and also an important reason why I have auto rotate off. The autoplay can fortunately be switched off, but if this behaviour does not become possibe I'll either install 0.19.8 again, or build a fork when I get the Rust stuff to compile in Android Studio.

**Comment by DuskyVoltage** [comment]
I cannot overstate how much I agree with @sanityormadness. I'm disabled and I have my tablet locked in portrait mode at all times (even if the device itself is horizontal) but watch videos in landscape. This used to be no problem at all, but now every time I try to watch a video in landscape it will change the entire app to landscape mode until I screw around with unlocking my device's orientation and rotating it then locking it again. 
Reverted to 0.19.8, I hope some kind of fix is made in the future.

**Comment by avently** [comment]
@opusforlife2 could you make a new project with important issues (for transition from old version to new one) that should be fixed ASAP? 
And maybe create a top level issue and link it to the github page. So every developer will see this project and maybe decide to fix some of the issues. Please, don't add them all, only important ones.
These issues I think important:
- can't go directly into fullscreen (this issue)
- #4479
- #4383
- #4403
- #4412
- and the old issue where we talked about new UI for buttons on top of thumbnail or instead of download button (`more`)

**Comment by wb9688** [comment]
@avently: I'd like to say that the issue that quite some users are reporting that the service is getting killed somehow, and thus stopping e.g. video playback, is also pretty important.

**Comment by avently** [comment]
@wb9688 yep, just write the issue number to @opusforlife2

**Comment by opusforlife2** [comment]
> could you make a new project with important issues (for transition from old version to new one) that should be fixed ASAP?

@avently It was decided in IRC that the 0.20.1 project will be used exactly for this purpose. You can open it and see.

> And maybe create a top level issue and link it to the github page.

I'll do that just now.

By the way, you joined the channel and left soon. What happened?

**Comment by avently** [comment]
@opusforlife2 I just wanted to see how active the channel is. I didn't see any messages after half of hour so I left it. I don't want to use IRC because it doesn't have a history I can see without following every time. I can see history using external services that save it

**Comment by opusforlife2** [comment]
@avently Join it using a Matrix account. That's what everyone usually does.

**Comment by Stypox** [comment]
~~I have tried multiple times, but I can't find a place where I can toggle fullscreen right after the player opened inside VideoDetailFragment. Both `onServiceConnect()` and `onPlaybackUpdate(STATE_BLOCKED)` are too early and do nothing. Btw, in order to toggle fullscreen I'm doing the same exact steps as if the user pressed on the fullscreen button, i.e.~~
```java
if (!isVerticalVideo
                || (service.isLandscape() && globalScreenOrientationLocked(service))) {
            fragmentListener.onScreenRotationButtonClicked();
        } else {
            toggleFullscreen();
        }
```
edit: `onPlaybackUpdate(STATE_BUFFERING)` works

@avently btw, as opus said you can join through Matrix, and the IRC channel is usually pretty active

**Comment by muzzol** [comment]
just want to add that my 3 years old son misses this feature so much and his father is tired of clicking in the little corner to put "PJ Masks" in full screen.

**Comment by barsupi** [comment]
I wonder if I'm the only one that doesn't use his.phone with auto rotate activated. i like it in forced portrait mode

Right now. The full screen button 1st change the phone orientation with no way to rotate it back without close thenapp cuz the button is hidden. And if I press again goes to full screen. Super annoying.

Please bring back the old behavior.
Press play and goes to full screen in landscape mode.

**Comment by doodhout** [comment]
Yes exactly, I really don't like that the video starts playing in a small viewport, I'd like it to start playing in fullscreen ALWAYS, so please include a setting to be able to use it like that, because it used to work like that before 0.20.0 and I'm rather fond of it.

**Comment by neitsab** [comment]
I'd like to add in support of this request that it isn't just a matter of one sole more tap:
- if autoplay is on, it is first one tap to display the on screen controls, then *at least* one more tap on the fullscreen icon, but much more likely 2 to 4+ because the icon is so small and if you hit the video instead it makes the controls disappear, so then it's rinse and repeat.
- if autoplay is off (my case), add one compulsory tap to the previous description.

It makes the experience really clumsy and tiring tens of times a day. Before I could reliably know that as soon as I tapped anywhere on the video thumbnail, my phone would go in landscape mode and I could start watching, no fuss. Now I am *expecting* the process to be clumsy and hard...

I should mention that I also have my phone locked in portrait mode, as the very only time I need it in landscape if when I watch videos, which made the previous behavior so appreciated. I have tried turning autorotate on again and couldn't believed how bothering it was: as soon as I slightly moved or needed a different orientation (in bed etc.) my whole UI (outside of NewPipe) would turn and twist back and forth... I locked back in portrait and don't see myself changing. Some phones definitely have finicky gyro sensors.

I am also perfectly fine with this option having to be toggled/set up on purpose in the Settings, no need to change the new defaults, but at least please offer a way to reinstate what was the default for so long. "Don't break the userspace" as they say in kernel land...

Thank you for your time and efforts in developping NewPipe.

**Comment by nikhilCad** [comment]
I feel like I am stupid : 
1)Open a video and click the little fullscreen button
2)Tap the video to get the title of the video shown
3) Click and drag the title down to the little miniplayer
4) Close the video using X button
5) Congratulations!  You are now locked in landscape. Enjoy your life.

**Comment by opusforlife2** [comment]
> 4) Close the video using X button

Personally, I think if your device is locked in portrait, step 4 should cause it to rotate back to portrait.

**Comment by Littlemac123** [comment]
https://user-images.githubusercontent.com/72648991/95651210-fab83a00-0ae8-11eb-8319-d7db378a14b7.png

I really want this "ZOOM" feature back so i can watch my phone in landscape mode ones again with out having to see that stupid tinny window at the top thats hard to see, zooming it in helps you see the middle of the video more clear and its useful with one hand

**Comment by Littlemac123** [comment]
it would be nice if someone could recompile the GUI back from v0.19.8 in to v0.20.2 and upload the APK

**Comment by doodhout** [comment]
@Littlemac123, I REALLY don't think it's that simple, only in utopia are UI and the logic behind the UI so detached and compatible they are fully interchangeable.

**Comment by opusforlife2** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/4686#issuecomment-719897601 guys.

**Comment by doodhout** [comment]
@opusforlife2 I stand corrected. 😅

**Comment by HattinGokbori87** [comment]
Really missing this feature. Couldn't stand this behaviour of the official client on mobile (on Tablets and desktops it makes sense) and switched to NewPipe the moment I came to know about it (apart from privacy related reasons). 
I personally can't find much importance of the info page anyway where it needs to be made default land-on page on NewPipe, especially considering you can't like the video and the comments section is also just a vanity function (not undermining work here. I know it's hard constantly battling with Google). All the info page does on a smartphone is playing the video in a tiny screen which is unwatchable unless you have a big-screen phone or on a tablet. I preferred the old behaviour of NewPipe where you can instantly start enjoying the video you want to start watching, and if you're interested in the detail information of the video or reading some comments you could always switch to portrait mode portrait very easily too. 
Current behaviour requires us suitor of the old ways and users of +-5 inch phones to either always go through at least extra 2 clicks (3 clicks if autoplay is turned off) for watching every video. Or turning on the autorotate function and be careful about not leaning on any sides while on bed. 
Would like to see the return of this feature (can be an option in the settings). 
Thanks to all putting effort in this project.

**Comment by mof22** [comment]
PLEASE restore the full screen reproduction in portrait mode.
In previous versions I can view a landscape video in portrait mode with the subtitles outside the video frame.
Now subtitles occupy an important part of the video box and frequently obstruct a relevant part of the image.
This is very useful for those who need the assistance of the subtitles when we do not understand the language of the video.
In my case I do not understand English when I hear it, but I can read it with some fluidity.

**Comment by opusforlife2** [comment]
@mof22 You've already made this same comment on the correct issue (4500), so I'm deleting it here. Please don't double post on different issues out of frustration (apologies if that is not the case). Open source projects progress at the pace the developers set when they have free time.

**Comment by mof22** [comment]
@opusforlife2
Sorry about that.
I did not know what the most suitable place was to put the comment.
Feel free to erase what looks inadequate.

**Comment by Littlemac123** [comment]
> https://user-images.githubusercontent.com/72648991/95651210-fab83a00-0ae8-11eb-8319-d7db378a14b7.png
> 
> I really want this "ZOOM" feature back so i can watch my phone in landscape mode ones again with out having to see that stupid tinny window at the top thats hard to see, zooming it in helps you see the middle of the video more clear and its useful with one hand

sadly this problem is still not here yet

**Comment by opusforlife2** [comment]
@Littlemac123 That has nothing to do with this issue. Please check other issues, or open a new one if you don't find a duplicate.
